src/Faithlife.Testing/AssertEx.cs

```diff
@@ -842,7 +836,47 @@ private static string ToString(object obj)
 			return toString;
 		}
 
-		private static readonly JsonSettings s_jsonWithTabs = new JsonSettings { IsIndented = true };
+		private static string ToPrettyJson(JToken token, int tabLevel)
+		{
+			var indent = new string('\t', tabLevel);
+			if (token.Type == JTokenType.Array)
+			{
+				var children = token.Children();
+				if (!children.Any())
+					return indent + "[]";
+
+				var childLines = children.Select(c => ToPrettyJson(c, tabLevel + 1)).AsReadOnlyList();
+				var shortJson = '[' + childLines.Select(l => l.Trim()).Join(", ") + ']';
+				if (shortJson.Length + tabLevel * 4 < c_maxJsonLength)
+					return indent + shortJson;
```

Comment:
You can avoid having to think about the `tabLevel * 4` stuff while also (IMO) communicating your intent more clearly by changing how you define `shortJson`:

```
				var shortJson = $"{indent}[{childLines.Select(l => l.Trim()).Join(", ")}]";
				if (shortJson.Length < c_maxJsonLength)
					return shortJson;
```

Reply:
I opted for the way I wrote it due to the fact that tab characters are 4 chars wide, but, of course, only count as one character in `.Length`.. But I could be convinced not to bother with counting tabs at all.

Reply:
Ah, I didn't think of that.

It *is* sort of odd that objects wrap more aggressively as we become more indented -- though that matches the intent of wrapping, which is to bound the horizontal space our output takes up in reasonable cases.

I could go either way.